applications/ParticleMechanicsApplication/custom_utilities/mpm_search_element_utility.h

```diff
@@ -429,8 +440,24 @@ namespace MPMSearchElementUtility
         std::vector<typename Element::Pointer> missing_elements;
         std::vector<typename Condition::Pointer> missing_conditions;
 
-        NeighbourSearchElements(rMPMModelPart, rBackgroundGridModelPart, missing_elements, Tolerance);
-        NeighbourSearchConditions(rMPMModelPart, rBackgroundGridModelPart, missing_conditions, Tolerance);
+        if (!rMPMModelPart.GetProcessInfo()[IS_RESTARTED]) {
+            NeighbourSearchElements(rMPMModelPart, rBackgroundGridModelPart, missing_elements, Tolerance);
+            NeighbourSearchConditions(rMPMModelPart, rBackgroundGridModelPart, missing_conditions, Tolerance);
+        } else {
+            #pragma omp parallel for
+            for (int i = 0; i < static_cast<int>(rMPMModelPart.Elements().size()); ++i) {
+                auto element_itr = (rMPMModelPart.ElementsBegin() + i);
+                #pragma omp critical
```

Comment:
this serializes the entire loop, making it even slower than without OMP :)

I recommend to directly use `std::vector<typename Condition::Pointer> = rMPMModelPart.Conditions()`. AFAIK this is not possible (due to type mismatch), hence the easiest and fastest solution is probably the following:
~~~c++
missing_elements.resize(rMPMModelPart.NumberOFElements());
IndexPartition(rMPMModelPart.NumberOFElements()).for_each([](std::size_t i){
    missing_elements[i] = &*(rMPMModelPart.ElementsBegin() + i); // maybe add/remove *&?
});

Reply:
True, good point.
I will try that solution and update, thank you! :)

Reply:
Ok done :thumbsup: